app/src/main/java/org/android/go/sopt/util/PreferenceUtil.kt

```diff
@@ -2,20 +2,43 @@ package org.android.go.sopt.util
 
 import android.content.Context
 import android.content.SharedPreferences
+import androidx.annotation.StringRes
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKey
 
-class PreferenceUtil(context: Context) {
-    private val prefs: SharedPreferences =
-        context.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE)
+class PreferenceUtil(private val context: Context) {
+    private val sharedPreferences: SharedPreferences by lazy {
+        // 암호화 할 마스터 키 생성
+        val masterKeyAlias = MasterKey
+            .Builder(context, MasterKey.DEFAULT_MASTER_KEY_ALIAS)
+            .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+            .build()
 
-    fun getString(key: String, defValue: String?): String? {
-        return prefs.getString(key, defValue)
+        // key, value를 암호화 한 sharedPreferences 객체 생성
+        EncryptedSharedPreferences.create(
+            context,
+            FILE_NAME,
+            masterKeyAlias,
+            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+        )
     }
 
-    fun setString(key: String, str: String) {
-        prefs.edit().putString(key, str).apply()
+    fun getString(@StringRes keyId: Int, defaultValue: String?): String? {
+        val key = context.getString(keyId)
+        return sharedPreferences.getString(key, defaultValue)
+    }
+
+    fun putString(@StringRes keyId: Int, newValue: String) {
+        val key = context.getString(keyId)
+        sharedPreferences.edit().putString(key, newValue).apply()
+    }
+
+    fun deleteAllData() {
+        sharedPreferences.edit().clear().apply()
```

Comment:
```suggestion
    fun putString(@StringRes keyId: Int, newValue: String) {
        val key = context.getString(keyId)
        sharedPreferences.edit() { putString(key, newValue) }
    }

    fun deleteAllData() {
        sharedPreferences.edit { clear() }
```

[Android KTX](https://developer.android.com/kotlin/ktx)를 활용하여 조금 더 코드를 경량화할 수도 있습니다!
[Google Developers 블로그 글 SharedPreferences 편집 부분](https://developers-kr.googleblog.com/2018/02/introducing-android-ktx-even-sweeter-kotlin-development-for-android.html)을 참고해주세요 :)